providers/default_rhel.rb

```diff
@@ -0,0 +1,136 @@
+  # Requirements
+  package node['kernel_modules']['packages']
+
+  # Module init loading section
+  template modload_file do
+    source "#{node['platform_family']}-#{node['platform_version'].to_i}/init-load.erb"
+    cookbook 'kernel-modules'
+    mode '0755'
+    variables(
+      name: new_resource.name,
+      bin: MODPROBE_BIN,
+    )
+    action new_resource.onboot ? :create : :delete
+  end
+
+  # Module loading options
+  template modprobe_file do
+    source 'modprobe.conf.erb'
+    cookbook 'kernel-modules'
+    mode '0644'
+    variables(
+      name: new_resource.name,
+      alias: new_resource.alias,
+      options: new_resource.options,
+      install: new_resource.install,
+      remove: new_resource.remove,
+      blacklist: new_resource.blacklist,
+    )
+    action :create
+  end
+end
+
+action :load do
+  # When loading a module it means the module must be configured
+  run_action :configure
+
+  wanted_action = blacklisted?(new_resource.install, new_resource.blacklist) ? :run : :nothing
+  execute "Unloading module #{new_resource.name}" do
+    command reload_cmd
+    only_if "lsmod | grep -q #{new_resource.name}"
+    only_if { new_resource.reload || new_resource.force_reload } # "force_reload" supersede "reload" property
+
+    subscribes :run, "template[#{modprobe_file}]" # 'Unload' only_if template has been updated
+    action wanted_action # If module blacklisted unload no matter what
+  end
+
+  execute "(Re)Loading module #{new_resource.name}" do
+    command "#{MODPROBE_BIN} -v #{new_resource.name}"
+    not_if "lsmod | grep -q #{new_resource.name}"
+    not_if { wanted_action.to_s == 'run' } # No need to load if blacklisted
+  end
+end
+
+action :unload do
+  # Requirements
+  package node['kernel_modules']['packages']
+
+  execute "Unloading module #{new_resource.name}" do
+    command reload_cmd
+    only_if "lsmod | grep -q #{new_resource.name}"
+    only_if { new_resource.reload || new_resource.force_reload } # "force_reload" supersede "reload" property
+  end
+end
+
+action :disable do
+  file modload_file do
+    action :delete
+  end
+end
+
+action :remove do
+  run_action :unload
+  [modload_file, modprobe_file].each do |f|
+    file f do
+      action :delete
+    end
+  end
+end
+
+# Function to return the correct commnd for removing module
+def reload_cmd
+  # First unload the module nicely
+  unload_cmd = MODPROBE_BIN + " -r #{new_resource.name}"
+  # If asked do it the strong way
+  unload_cmd + " || /sbin/rmmod -f #{new_resource.name}" if new_resource.force_reload
+  unload_cmd
+end
+
+def modload_file
+  case node['platform_version']
+  when /^6/
+    ::File.join(node['kernel_modules']['modules_load.d'], new_resource.name + '.modules')
+  when /^7/
+    ::File.join(node['kernel_modules']['modules_load.d'], new_resource.name + '.conf')
+  end
+end
+
+def modprobe_file
+  ::File.join(node['kernel_modules']['dirs']['modprobe.d'], new_resource.name + '.conf')
+end
+
+def blacklisted?(install, blacklist)
```

Comment:
Why use arguments here instead of using the `new_resource `  object just like in other methods?

If you name this method `blacklisted?` then you can remove the `wanted_action` variable which is confusing too